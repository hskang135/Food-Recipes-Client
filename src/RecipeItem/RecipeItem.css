body {
  text-align: center;
  background-color: #eeecec;
}

.recipeItem {
  list-style:none;
  margin-right: 2.1rem;
  font-size: 1.2rem;
  font-family: 'Nunito', sans-serif;
}

.recipeItem_info {
  border: 0.1rem solid black;
  max-width: 46rem;
  padding-right: 1.2rem;
  box-shadow: 2px 2px 5px #424141;
}

button {
  background-color: #2a8a93;
  color: white;
  padding: 0.6rem;
  font-size: 1.1rem;
  margin-top: 0.5rem;
  margin-bottom: 2rem;
  text-align: center;
  font-family: 'Nunito', sans-serif;
}

li {
  padding-bottom: 1rem;
  text-align: left;
  list-style-type: square;
}

@media screen and (min-width: 1024px) {
  .recipeItem {
    margin: auto;
    font-size: 1.6rem;
    margin-bottom: 2.5rem;
  }

  .recipeItem_info {
    margin: auto;
    max-width: 52rem;
  }

  button {
    padding: 0.8rem;
    font-size: 1.5rem;
    margin-top: 1.2rem;
    margin-bottom: 2.5rem;
  }
}

@media screen and (min-width: 1025px) {
  button {
    padding: 0.8rem;
    font-size: 1.5rem;
    margin-top: 1.2rem;
    margin-bottom: 2.8rem;
  }

  .recipeItem_info {
    margin: auto;
    max-width: 56rem;
  }
}


