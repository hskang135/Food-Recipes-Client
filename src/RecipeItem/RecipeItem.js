import React from 'react';
import RecipesContext from '../RecipesContext';
import config from '../config';

function deleteRecipeRequest(id, cb) {
  fetch(config.API_ENDPOINT + `/recipes/${id}`, {
    method: 'DELETE',
    headers: {
      'content-type': 'application/json'
    }
  })
    .then(res => {
      if(!res.ok) {
        return res.json().then(error => Promise.reject(error))
      }
    })
    .then(() => {
      cb(id)
    })
    .catch(error => {
      console.error(error)
    })
};

export default function RecipeItem(props) {
  return (
      <RecipesContext.Consumer>
        { (context) => (
          <li className ='RecipeItem'>
            <div className = 'RecipeItem_row'>
              <h2>{props.foodname}</h2>
              
              <ul>
                <li>Ingredients: {props.ingredients}</li>
                <li>Description: {props.description}</li>
              </ul>
            </div>

            <div className='recipesButton'>
              <button className='delete'
                onClick={() => {
                  deleteRecipeRequest(
                    props.id,
                    context.deleteRecipe
                  )
                }}> 
                Delete Recipe
              </button>
            </div>
          </li>
        )}
      </RecipesContext.Consumer>
  )


};





