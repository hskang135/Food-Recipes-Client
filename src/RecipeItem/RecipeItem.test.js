import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import RecipeItem from './RecipeItem'

describe(`RecipeItem component`, () => {
  const props = {
    id: 'a11114yjfj',
    foodname: 'test-class-name',
    ingredients: 'test-ingredients',
    description: 'test-description'
  }

  it('renders Recipes by default', () => {
    const wrapper = shallow(<RecipeItem />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders Recipes given props', () => {
    const wrapper = shallow(<RecipeItem {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})


