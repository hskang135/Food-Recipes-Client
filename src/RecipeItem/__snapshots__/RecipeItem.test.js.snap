// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeItem component renders Recipes by default 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`RecipeItem component renders Recipes given props 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
